### DISCUSSION TITLE
Performance considerations: using meta.value vs control.value in Switch component example.

### DISCUSSION THREAD

**Discussion by reborn2209** [edmundhung/conform#718] [Show and tell]
## Seeking Clarification on Switch Component Implementation

I hope this message finds you well. I've been studying the excellent examples provided in the conform repository, and I came across an interesting point in the [shadcn Switch component example](https://github.com/edmundhung/conform/blob/main/examples/shadcn-ui/src/components/conform/Switch.tsx#L17). I noticed that `meta.value` is being used directly, and I'm curious about the reasoning behind this choice.

During my testing with React DevTools, I observed that the entire form was re-rendering each time the switch state changed. This observation led me to wonder if there might be potential for optimization.

## Current Implementation

The Switch component currently uses `meta.value`:

```tsx
checked={meta.value === "on"}

## Altenative implementation
```tsx
checked={control.value === "on"}
```

**Comment by edmundhung** [comment]
Using `control.value` is probably a better approach here. Accessing the value on metadata is generally more expensive and you will need it mostly when you don't have direct access to the control value.

It would be awesome if you can help putting up a PR. Thanks!

**Reply by reborn2209** [reply]
Thank you for your prompt and insightful feedback. 

I appreciate your guidance on using control.value instead of meta.value for better performance. 
As per your suggestion, I have created a pull request implementing this change. I'm grateful for the opportunity to contribute to conform

https://github.com/edmundhung/conform/pull/721
